src/NetCore/RequestTools.cs

```diff
@@ -37,7 +37,7 @@ internal static bool Validate(HttpRequest req, ISigningAlgorithm algorithm, IApp
                 var builder = new CannonicalRepresentationBuilder();
 
                 // Handling request possibly coming from rewrite
-                var url = h.Get(Headers.XOriginalUrl); // Original URL before rewrite
+                var url = url_ ??  h.Get(Headers.XOriginalUrl); // Original URL before rewrite
```

Comment:
You can't use **static** URL as validation will not work. Configuration must have ability to specify **Public Host** or something similar.